hazelcast/src/main/java/com/hazelcast/internal/cluster/ClusterService.java

```diff
@@ -162,8 +162,8 @@
      *    <li>When there is a split-brain issue, if further splits occur in the already split sub-clusters, the
      *    uniqueness guarantee can be lost.
      * </ul>
-     * The value returned from this method can be cached. Even though it can change later, both values are
-     * unique.
+     * When duplicate member list join version is assigned during network split, the returned value can
```

Comment:
I am not sure about this new part. We already document the failure scenarios here. If the uniqueness lost, safety is already broken. The caller has no way to verify uniqueness by just calling this method. When the returned member list join version changes, I think still it may not be unique if it is bigger than a 2-split (i.e, there are more than 2 sub-clusters after split).

Reply:
This is about damage control. Before it stated that we can cache it forever which will make the member to generate duplicate IDs until it is restarted. Now if a duplicate happens (which it can), the situation can recover automatically and we'll stop generating duplicates.